File: src/main/java/tax/bilibili/nineblog/entity/RestResponse.java
package tax.bilibili.nineblog.entity;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import tax.bilibili.nineblog.Enum.Http.HttpMessage;
import tax.bilibili.nineblog.Enum.Http.HttpStatus;

/**
 * RestResponse
 *
 * @author sheip9
 * @since 2023/10/22 19:31
 */
@Data
@NoArgsConstructor
@AllArgsConstructor
public class RestResponse<T> {
    private int status;
    private String message;
    private T data;

    public RestResponse(HttpStatus status, HttpMessage message, T data) {
        this.status = status.getCode();
        this.message = message.getValue();
        this.data = data;
    }

    public static <T> RestResponse<T> ok(T data) {
        return new RestResponse<>(HttpStatus.Ok, HttpMessage.SUCCESS, data);
    }

    public static <T> RestResponse<T> ok() {
        return new RestResponse<>(HttpStatus.Ok, HttpMessage.SUCCESS, null);
    }
}

File: src/main/java/tax/bilibili/nineblog/model/RestResponse.kt
package tax.bilibili.nineblog.model

import com.fasterxml.jackson.annotation.JsonInclude
import org.springframework.http.HttpStatus
import java.io.Serializable

@JsonInclude(JsonInclude.Include.NON_NULL)
data class RestResponse<T>(
    val status: Int,
    val message: Serializable,
    val data: T?,
) : Serializable {
    constructor(status: Int,message: Serializable) : this(status, message, null)
    constructor(status: HttpStatus,message: Serializable) : this(status.value(), message, null)
    constructor(status: HttpStatus,message: Serializable, data: T?) : this(status.value(), message, data)
}
